core/src/main/java/org/elasticsearch/index/mapper/core/DateFieldMapper.java

```diff
@@ -125,15 +126,19 @@ public DateFieldMapper build(BuilderContext context) {
         }
 
         protected void setupFieldType(BuilderContext context) {
-            FormatDateTimeFormatter dateTimeFormatter = fieldType().dateTimeFormatter;
-            // TODO MOVE ME OUTSIDE OF THIS SPACE?
             if (Version.indexCreated(context.indexSettings()).before(Version.V_2_0_0)) {
-                boolean includesEpochFormatter = dateTimeFormatter.format().contains("epoch_");
+                if (fieldType().dateTimeFormatter().equals(Defaults.DATE_TIME_FORMATTER)) {
```

Comment:
added a new commit with a test fixing this. I just check if the the parser actually parsed a new format and only in case if not, the defaults are applied